### ISSUE TITLE
Support figwheel out of the box

### ISSUE THREAD

**Issue by aaronjensen** [tpope/vim-fireplace#213]
[Figwheel is capable of launching a repl]() and people are using [hacks](https://gist.github.com/malyn/646e291d01101dcc8b48) to get vim-fireplace to work. The hack itself didn't work for me, I get:

```
IllegalArgumentException No value supplied for key: {doc #<repl$doc_help figwheel_sidecar.repl$doc_help@6650e34e>, import #<repl$fn__4279$self__4300 cljs.repl$fn__4279$self__4300@543d18
76>, fig-status #<repl$make_special_fn$self__17812 figwheel_sidecar.repl$make_special_fn$self__17812@d7bff2f>, stop-autobuild #<repl$make_special_fn$self__17812 figwheel_sidecar.repl$ma
ke_special_fn$self__17812@69d7dda0>, in-ns #<repl$fn__4279$self__4285 cljs.repl$fn__4279$self__4285@38f36c75>, start-autobuild #<repl$make_special_fn$self__17812 figwheel_sidecar.repl$m
ake_special_fn$self__17812@6aa36ec3>, build-once #<repl$make_special_fn$self__17812 figwheel_sidecar.repl$make_special_fn$self__17812@3dce94ea>, switch-to-build #<repl$make_special_fn$s
elf__17812 figwheel_sidecar.repl$make_special_fn$self__17812@441092a1>, require #<repl$fn__4279$self__4291 cljs.repl$fn__4279$self__4291@160771c3>, clojure.core/load-file #<repl$fn__427
9$self__4281 cljs.repl$fn__4279$self__4281@1f62dae2>, load-namespace #<repl$fn__4279$self__4306 cljs.repl$fn__4279$self__4306@4be8aca1>, load-file #<repl$fn__4279$self__4281 cljs.repl$f
n__4279$self__4281@1f62dae2>, clojure.core/in-ns #<repl$fn__4279$self__4285 cljs.repl$fn__4279$self__4285@38f36c75>, add-dep #<repl$add_dep figwheel_sidecar.repl$add_dep@23e3af34>, clea
n-builds #<repl$make_special_fn$self__17812 figwheel_sidecar.repl$make_special_fn$self__17812@272da14a>, reset-autobuild #<repl$make_special_fn$self__17812 figwheel_sidecar.repl$make_sp
ecial_fn$self__17812@292e9caa>, require-macros #<repl$fn__4279$self__4296 cljs.repl$fn__4279$self__4296@432b0b3f>}  clojure.lang.PersistentHashMap.create (PersistentHashMap.java:77)
```

Could you add, or please point in the direction of adding proper support for figwheel? Thanks!

**Comment by tpope** [comment]
You could make that hack conditional by checking for `!empty(fireplace#findresource('cemerick/piggieback.clj'))` to determine the availability of a given library.

**Comment by tpope** [comment]
As for getting the hack itself working I have no idea. That's on you to figure out the correct call.

**Comment by aaronjensen** [comment]
I was able to get this working, one of the big keys was using a not-yet-released version of weasel 0.7.0-SNAPSHOT and figwheel-sidecar instead of lein-figwheel and `:Piggieback (weasel.repl.websocket/repl-env :ip "0.0.0.0" :port 9001)` (no hack needed)

If you're interested, I could write it up for a wiki/readme or just in this issue, but I can consider this closed, it vim-fireplace does support figwheel out of the box, it's just not obvious how to do it.

**Comment by jcw** [comment]
_Apologies for commenting on a closed issue, but I think I'm running into something similar..._

I've installed "tpope/vim-fireplace" + "guns/vim-clojure-static" with Vundle (via closure-doc.org tutorial) - works great. I've also created new projects with `lein new figwheel ...` and `lein new chestnut ...` - both work great as well.

Is there a way to get fireplace's easy doc lookup and jump-to-definition working with a figwheel setup?

I understand that there can be a slight mismatch, as figwheel reloads on save, so there's less need to reload directly via the REPL, but I'd like to use fireplace's doc, jumps, in-context eval working, if possible.

I'm very new to the Closure(Script) game, but eager to learn, and I'd be grateful for any tips or pointers.
